src/server/transaction.cc

```diff
@@ -1370,27 +1370,26 @@ bool Transaction::IsGlobal() const {
 // Returns true if the transacton has changed its state from suspended to awakened,
 // false, otherwise.
 bool Transaction::NotifySuspended(TxId committed_txid, ShardId sid, string_view key) {
-  unsigned idx = SidToId(sid);
-  auto& sd = shard_data_[idx];
-  unsigned local_mask = sd.local_mask;
-
-  if (local_mask & Transaction::EXPIRED_Q) {
-    return false;
-  }
-
   // Wake a transaction only once on the first notify.
   // We don't care about preserving the strict order with multiple operations running on blocking
   // keys in parallel, because the internal order is not observable from outside either way.
   if (wakeup_requested_.fetch_add(1, memory_order_relaxed) > 0)
     return false;
 
-  DVLOG(1) << "NotifySuspended " << DebugId() << ", local_mask:" << local_mask << " by commited_id "
-           << committed_txid;
+  DCHECK(coordinator_state_ & COORD_BLOCKED);  // We wake only once, so it must be still blocked
```

Comment:
`coordinator_state_` is modified by the coordinator thread. Could it be that it's been expired and modified in parallel?